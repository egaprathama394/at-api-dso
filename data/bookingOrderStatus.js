export default {
  orderDaimoCode: 'DFA202000000123',
  pssOrderStatus: '2',
};
