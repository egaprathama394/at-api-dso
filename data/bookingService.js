const randomstring = require('randomstring');

const fullNameMax = randomstring.generate({
  charset: 'string',
  length: '110',
});

const noPhoneMax = randomstring.generate({
  charset: 'numeric',
  length: '17',
});

const noPhoneRandom = randomstring.generate({
  charset: 'numeric',
  length: '18',
});

const fullNameRandom = randomstring.generate({
  charset: 'string',
  length: '111',
});

const randomNopol1 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol2 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol3 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

const randomNopol4 = randomstring.generate({
  charset: 'numeric',
  length: '4',
});

export default {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: '',
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data submit booking service',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: '',
  policeRegNo: `AB${randomNopol1}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for submit booking service max lenght

export const maxInput = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: `${fullNameMax}`,
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data booking service dengan inputan max char',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: `081${noPhoneMax}`,
  policeRegNo: `AC${randomNopol2}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for submit booking service max lenght + 1

export const negativeRequest = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: `${fullNameRandom}`,
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data submit booking service dengan inputan max + 1 char',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: `081${noPhoneRandom}`,
  policeRegNo: `AD${randomNopol3}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

// data for cancel booking service

export const inputDataForCancel = {
  address: '',
  bookingDateTime: '2020-04-01 10:00:00',
  branchCode: 'D004',
  city: 'DKI JAKARTA',
  fullName: '',
  jobService: [
    {
      actual: 0.0,
      id: 0,
      isActive: false,
      isPickedJob: false,
      jobCode: '10000',
      jobDescription: 'CHECKING 10.000',
      price: 0,
      rate: 0,
    },
  ],
  notes: 'Data untuk cancel booking service',
  orderEndTime: '2020-04-01 15:00:00',
  orderStartTime: '2020-04-01 10:00:00',
  phoneNo: '',
  policeRegNo: `AA${randomNopol4}XYZ`,
  vehicleType: 'LUXIO',
  vin: '',
  year: 2014,
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

export const cancelRequest = {
  orderNo: '',
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

// data for display booking service

export const displayBookingService = {
  isWalkIn: false,
  orderBy: '',
  AccountType: '1',
  Action: '',
  ApplicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};

export const invalidDisplayBookingService = {
  isWalkIn: '',
  orderBy: '',
  AccountType: '',
  Action: '',
  ApplicationName: '',
  DeviceType: '',
  OperatingSystem: '',
  UserVersion: '',
};
