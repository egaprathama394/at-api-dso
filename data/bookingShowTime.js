export default {
  branchCode: 'D004',
  scheduleDate: '2020-09-30 00:00:00',
  vehicleType: 'XENIA MT',
  jobs: [
    {
      jobCode: 'OTHER',
      jobDescription: 'PERBAIKAN LAINNYA',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};

export const negativeRequest = {
  branchCode: 'D004',
  scheduleDate: '2018-10-26 00:00:00',
  vehicleType: 'XENIA MT',
  jobs: [
    {
      jobCode: 'OTHER',
      jobDescription: 'PERBAIKAN LAINNYA',
    },
  ],
  action: '',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
