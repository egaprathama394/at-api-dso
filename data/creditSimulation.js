export default {
  P_NEW_USED: 'N',
  P_CATEGORY: 'EX2',
  P_VEHICLE_CATEGORY: '1',
  P_BRAND: '002',
  P_TYPE: '1',
  P_MODEL: '1',
  P_DP: '49125000',
  P_OTR: '196500000',
  P_TYPE_INSU: 'A',
  P_MODE_INSU: 'K',
  P_FLAG_ACP: 'Y',
  P_TENORVALUE: '12',
};