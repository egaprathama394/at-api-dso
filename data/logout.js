export default {
  applicationName: '8EA6E777-AA05-4CAF-9F54-5B6A106760E2',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
  deviceID: '8B55A4DA-5331-47F9-9C8D-838A110FA18C',
};


export const negativeRequest = {
  action: 'Display',
  applicationName: '9093D285-5263-4B9A-97D7-C185189B0695',
  userVersion: '1.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
};
