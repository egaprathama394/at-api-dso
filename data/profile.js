export default {
  action: 'Display',
  applicationName: '9093D285-5263-4B9A-97D7-C185189B0695',
  userVersion: '1.0.0',
  deviceType: '0',
  operatingSystem: '0',
  accountType: '1',
  deviceID: '74eef5b5-69c0-454d-9e2d-485c9e988766',
};

export const editProfile = {
  userAccount: '',
  AccountType: '1',
  Action: 'Edit',
  ApplicationName: '35AE57AD-572F-4EAF-B908-3B0523ADEDB8',
  DeviceType: '0',
  OperatingSystem: '0',
  UserVersion: '1.0.0',
};
