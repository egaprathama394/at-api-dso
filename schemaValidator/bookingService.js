export default {
  type: 'object',
  required: [
    'orderNo',
    'acknowledge',
    'message',
    'version',
    'build',
  ],
  properties: {
    orderNo: {
      type: 'string',
    },
    acknowledge: {
      type: 'integer',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const displayBookingService = {
  type: 'object',
  required: [
    'bookingOrderList',
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    bookingOrderList: {
      type: 'array',
      items: {
        type: 'object',
        required: [
          'branchCode',
          'branchPhone1',
          'branchPhone2',
          'branchDescription',
          'branchAddress',
          'branchPICName',
          'orderNo',
          'orderStatus',
          'orderType',
          'totalAmount',
          'serviceOrderJobs',
          'orderStartTime',
          'orderEndTime',
          'policeRegNo',
          'vin',
          'vehicleType',
          'year',
          'name',
          'phone1',
          'phone2',
          'address',
          'serviceAddress',
          'notes',
          'dscProcessStatus',
          'actualFinishTime',
          'isEligibleForRating',
          'createdTime',
        ],
        properties: {
          branchCode: {
            type: 'string',
          },
          branchPhone1: {
            type: 'string',
          },
          branchPhone2: {
            type: 'string',
          },
          branchDescription: {
            type: 'string',
          },
          branchAddress: {
            type: 'string',
          },
          branchPICName: {
            type: 'string',
          },
          orderNo: {
            type: 'string',
          },
          orderStatus: {
            type: 'integer',
          },
          orderType: {
            type: 'integer',
          },
          totalAmount: {
            type: 'integer',
          },
          serviceOrderJobs: {
            type: 'array',
            items: {
              type: 'object',
              required: [
                'jobCode',
                'jobDescription',
              ],
              properties: {
                jobCode: {
                  type: 'string',
                },
                jobDescription: {
                  type: 'string',
                },
              },
            },
          },
          orderStartTime: {
            type: 'string',
          },
          orderEndTime: {
            type: 'string',
          },
          policeRegNo: {
            type: 'string',
          },
          vin: {
            type: 'string',
          },
          vehicleType: {
            type: 'string',
          },
          year: {
            type: 'integer',
          },
          name: {
            type: 'string',
          },
          phone1: {
            type: 'string',
          },
          phone2: {
            type: 'string',
          },
          address: {
            type: 'string',
          },
          serviceAddress: {
            type: 'string',
          },
          notes: {
            type: 'string',
          },
          dscProcessStatus: {
            type: 'integer',
          },
          actualFinishTime: {
            type: 'string',
          },
          isEligibleForRating: {
            type: 'boolean',
          },
          createdTime: {
            type: 'string',
          },
        },
      },
    },
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const negativeRequest = {
  type: 'object',
  required: [
    'acknowledge',
    'version',
    'build',
  ],
  properties: {
    acknowledge: {
      type: 'integer',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
  },
};

export const invalidRequest = {
  type: 'object',
  required: [
    'message',
    'modelState',
  ],
  properties: {
    message: {
      type: 'string',
    },
    modelState: {
      type: 'object',
      required: [
        'request.OrderBy',
      ],
      properties: {
        'request.OrderBy': {
          type: 'array',
          items: {
            type: 'string',
          },
        },
      },
    },
  },
};
