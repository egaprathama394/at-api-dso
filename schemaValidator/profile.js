export default {
  properties: {
    userAccount: {
      type: 'object',
      properties: {
        email: {
          type: 'string',
        },
        firstName: {
          type: 'string',
        },
        lastName: {
          type: 'string',
        },
        phoneNo: {
          type: 'string',
        },
        address: {
          type: 'string',
        },
        provinceCode: {
          type: 'string',
        },
        districtCode: {
          type: 'string',
        },
        subdistrictCode: {
          type: 'string',
        },
        villageCode: {
          type: 'string',
        },
        zipCode: {
          type: 'string',
        },
        birthdate: {
          type: 'string',
        },
        gender: {
          type: 'integer',
        },
        religion: {
          type: 'integer',
        },
        hobby: {
          type: 'string',
        },
        location: {
          type: 'string',
        },
        identificationType: {
          type: 'string',
        },
        identificationNumber: {
          type: 'string',
        },
        identificationDate: {
          type: 'string',
        },
        urlPhotoProfile: {
          type: 'string',
        },
        occupation: {
          type: 'string',
        },
        defaultPaymentMethod: {
          type: 'integer',
        },
        referralCode: {
          type: 'string',
        },
      },
      required: [
        'email',
        'firstName',
        'lastName',
        'phoneNo',
        'address',
        'provinceCode',
        'districtCode',
        'subdistrictCode',
        'villageCode',
        'zipCode',
        'birthdate',
        'gender',
        'religion',
        'hobby',
        'location',
        'identificationType',
        'identificationNumber',
        'identificationDate',
        'urlPhotoProfile',
        'occupation',
        'defaultPaymentMethod',
        'referralCode',
      ],
    },
    acknowledge: {
      type: 'integer',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
    errorCode: {
      type: 'number',
    },

  },
  required: [
    'userAccount',
    'acknowledge',
    'message',
    'version',
    'build',
  ],
  additionalProperties: false,
};

export const invalidToken = {
  properties: {
    error: {
      type: 'string',
    },
    error_description: {
      type: 'string',
    },
  },
  required: [
    'error',
    'error_description',
  ],
  additionalProperties: false,

};

export const editProfileSchema = {
  properties: {
    acknowledge: {
      type: 'number',
    },
    message: {
      type: 'string',
    },
    version: {
      type: 'string',
    },
    build: {
      type: 'string',
    },
    errorCode: {
      type: 'number',
    },
  },
  required: [
    'acknowledge',
    'message',
    'version',
    'build',
    'errorCode',
  ],
  additionalProperties: false,
};