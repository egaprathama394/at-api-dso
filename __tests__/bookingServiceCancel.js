/* eslint-disable max-len */
import getProfile from '../endpoints/profile';
import * as profileSchema from '../schemaValidator/profile';
import * as body from '../data/profile';
import submitBookingService from '../endpoints/bookingService-Submit';
import * as BookingServiceSchema from '../schemaValidator/bookingService';
import * as bodyBookingService from '../data/bookingService';
import cancelBookingService from '../endpoints/bookingService-cancel';

const fs = require('fs');
const { matchers } = require('jest-json-schema');

expect.extend(matchers);

const ambilToken = fs.readFileSync('./token.json');
const objToken = JSON.parse(ambilToken);

const token = objToken.login;

describe('Cancel booking service', () => {
  test('Can cancel booking service successfully using authorized token - positive test cases', async () => {
    const response = await getProfile(token, body.default);
    expect(response.status).toEqual(200);
    expect(response.body).toMatchSchema(profileSchema.default);
    const fullName = await response.body.userAccount.firstName.concat(' ', response.body.userAccount.lastName);
    bodyBookingService.inputDataForCancel.fullName = await fullName;
    bodyBookingService.inputDataForCancel.phoneNo = await response.body.userAccount.phoneNo;
    const responseSubmit = await submitBookingService(token, bodyBookingService.inputDataForCancel);
    expect(responseSubmit.status).toEqual(200);
    expect(responseSubmit.body).toMatchSchema(BookingServiceSchema);
    expect(responseSubmit.body.message).toEqual('Booking Service Reservation has been submitted');
    bodyBookingService.cancelRequest.orderNo = await responseSubmit.body.orderNo;
    const responseCancel = await cancelBookingService(token, bodyBookingService.cancelRequest);
    expect(responseCancel.status).toEqual(200);
    expect(responseCancel.body).toMatchSchema(BookingServiceSchema);
    expect(responseCancel.body.message).toEqual(`Booking dengan nomor ${responseSubmit.body.orderNo} telah dibatalkan`);
  });

  test('can not cancel booking service if orderNo not valid - negative test cases', async () => {
    bodyBookingService.cancelRequest.orderNo = await 'xxx';
    const responseCancel = await cancelBookingService(token, bodyBookingService.cancelRequest);
    expect(responseCancel.status).toEqual(200);
    expect(responseCancel.body).toMatchSchema(BookingServiceSchema);
    expect(responseCancel.body).toMatchSnapshot();
  });
});
