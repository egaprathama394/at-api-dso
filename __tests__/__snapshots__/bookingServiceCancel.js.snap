// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cancel booking service can not cancel booking service if orderNo not valid - negative test cases 1`] = `
Object {
  "acknowledge": 0,
  "build": "0",
  "message": "Order No tidak valid
",
  "orderNo": "",
  "version": "1.0",
}
`;